src/main/java/com/springml/salesforce/wave/impl/BulkAPIImpl.java

```diff
@@ -88,9 +88,11 @@ public boolean isCompleted(String jobId) throws Exception {
         if (batchInfos != null) {
             for (BatchInfo batchInfo : batchInfos) {
                 LOG.debug("Batch state : " + batchInfo.getState());
-                isCompleted = STR_COMPLETED.equals(batchInfo.getState());
+                isCompleted = STR_COMPLETED.equals(batchInfo.getState()) || STR_NOT_PROCESSED.equals(batchInfo.getState());
```

Comment:
* Need an inline comment to link to the `Not Processed` API doc: https://developer.salesforce.com/docs/atlas.en-us.api_asynch.meta/api_asynch/asynch_api_batches_interpret_status.htm. Because it is not straightforward to say "Not Processed" is `isCompleted`.
* I saw in the API doc: "After the original batch is changed to this state, you can monitor the subsequent batches and retrieve each batch’s results when it’s completed." Do we get the subsequent batch IDs?

Reply:
1) Done.
2) Yes, when we fetch for the batches within a job, it'll include all the subsequent batches described in the API doc. We will ignore the `Not Processed` batch and check for the completion of the subsequent batches.